### ISSUE TITLE
About t schedule during training

### ISSUE THREAD

**Issue by penghao-wu** [ZhengrongYue/UniFlow#3]
Hi, thanks for your interesting work. 
I notice that you use the one-step Euler at inference, could you please clarify the t sampling schedule used during training? Thanks~

**Comment by ZhengrongYue** [comment]
Hi, @penghao-wu, thank you for your interest!
We utilize the **fat_lognormal** schedule for sampling the timestep $t$ when training the Rectified Flow. This approach focuses the training efforts by primarily sampling from a Sigmoid-transformed Normal distribution. Simultaneously, the strategy incorporates a "fat tail" by occasionally introducing uniform sampling across the entire $t \in [0, 1]$ interval.

**Comment by penghao-wu** [comment]
@ZhengrongYue 
Hi, I am still a bit confused, as the sigmoid of the standard normal distribution should concentrate on 0.5 instead of 0. What is the mean of your normal distribution?

**Comment by ZhengrongYue** [comment]
Hi, @penghao-wu, I sincerely apologize for the confusing statement. You are right.The distribution of $t = \text{sigmoid}(nt)$ (where $nt \sim \mathcal{N}(0, 1)$) is a Logit-Normal distribution, which concentrates around $0.5$, not $0$.

```
nt = torch.randn((b*n_vis,)).to(x1.device)
t = torch.sigmoid(nt)
t = torch.where(torch.rand_like(t) <= 0.9, t, torch.rand_like(t))
```

**Comment by penghao-wu** [comment]
@ZhengrongYue I still have a few questions about the training details that I’d like to confirm with the authors:

1. In **Table 6**, the AdamW `beta1` is listed as **0.5** — is this a typo (should it be **0.9**)? Also, could you clarify what the **weight decay** value was?
2. In **Table 6**, the batch size is shown as **256**, but in **Section 4.1** it’s mentioned as **512**. Which value was actually used in training?
3. For the **self-distillation loss**, were the features *before* the transformer blocks (i.e., after only the embedding layer) included in the loss? And were the **CLS tokens** also included?

I tried reproducing the results using the current settings from the paper (`beta1=0.9`, `bs=256`), but my results are slightly worse (**FID ≈ 0.5, PSNR ≈ 32**).
Are there any other important training details or hyperparameter differences compared to what’s written in the paper that I should pay attention to?

Thanks a lot!

**Comment by ZhengrongYue** [comment]
Hi @penghao-wu, many thanks for the questions.
1.the β1 printed as 0.5 was a typo and should read 0.9; we did not use weight decay.
2.The global batch size is 512 (32 A800 × 16 per GPU), so the “256” that slipped into Table 6 will be corrected in the next arXiv revision.
3.We distill both the CLS token and all patch tokens from the output of every transformer block in the encoder.
We’re not certain whether the smaller batch hurts reconstruction, but we did observe that the flow loss converges slightly more slowly with batch 256. Additionally, for the flow head please make sure to follow FlowMo’s training and sampling strategy. During training, the timestep t must be sampled independently for each token. Hope this helps!

**Comment by penghao-wu** [comment]
Thanks for your answers, Zhengrong!
